docs/proposals/ose-rebase.md

```diff
@@ -21,16 +21,155 @@ There are different types of commits that need to be carried on top of Origin ma
+1abdc18 [CARRY][BRANDING] Branding updates
+```
+Shortly after Bob's PR is merged, Jane updates her fork and prepares an update for branding
+code which also needs to be carried long-term:
+```sh
+$ git add pkg/bar.go pkg/baz.go
+$ git commit -m "[CARRY][BRANDING] Branding changes in the oauth templates"
+```
+Now in Jane's branch, the following set of carries exist:
+```
+ratmats [CARRY][BRANDING] Branding changes in the oauth templates
+fvda4fs [SQUASH][BRANDING] Updated branding code
+aaed106 Automatic commit of package [atomic-openshift] release [3.6.17-1].
+6a27890 [CARRY][BUILD_GEN] Specfile updates
+9bcbe1d [CARRY][BUILD] Tooling updates
+1abdc18 [CARRY][BRANDING] Branding updates
+```
+Notice that there are two `[CARRY]` commits with the same type (BRANDING). This change is
+invalid based on [rule #5](#naming-requirements). Jane revisits this doc and updates her
+commit message to:
+```
+[SQUASH][BRANDING] Branding changes in the oauth templates
+```
+The commit name is now valid and the PR can be merged thus resulting in the following set
+of carried commits in OSE master:
+```
+sge53da [SQUASH][BRANDING] Branding changes in the oauth templates
+fvda4fs [SQUASH][BRANDING] Updated branding code
+aaed106 Automatic commit of package [atomic-openshift] release [3.6.17-1].
+6a27890 [CARRY][BUILD_GEN] Specfile updates
+9bcbe1d [CARRY][BUILD] Tooling updates
+1abdc18 [CARRY][BRANDING] Branding updates
+```
+Note that you can have as many `SQUASH` commits with the same type as possible.
+
+Bob, again, needs to check in a new script specific to OSE so he creates a new commit with
+the script:
+```sh
+$ git add hack/set-foo.sh
+$ git commit -m "[SQUASH][SCRIPT] script for setting OSE-specific envs"
+```
+The new set of carries in Bob's branch is:
+```
+d4sf5fg [SQUASH][SCRIPT] script for setting OSE-specific envs
+sge53da [SQUASH][BRANDING] Branding changes in the oauth templates
+fvda4fs [SQUASH][BRANDING] Updated branding code
+aaed106 Automatic commit of package [atomic-openshift] release [3.6.17-1].
+6a27890 [CARRY][BUILD_GEN] Specfile updates
+9bcbe1d [CARRY][BUILD] Tooling updates
+1abdc18 [CARRY][BRANDING] Branding updates
+```
+Bob realizes just before he starts the test job for his PR that he got it wrong based on
+[rule #4](#naming-requirements). He steps back for a moment to think if his commit needs
+to squash into a pre-exisiting `[CARRY]` commit or deserves to be its own `[CARRY]`. After
+discussing it with Jane, they conclude that the new script can be squashed in the
+`[CARRY][BUILD]` commit. Thus, the state of master ends up being
+```
+d4sf5fg [SQUASH][BUILD] script for setting OSE-specific envs
+sge53da [SQUASH][BRANDING] Branding changes in the oauth templates
+fvda4fs [SQUASH][BRANDING] Updated branding code
+aaed106 Automatic commit of package [atomic-openshift] release [3.6.17-1].
+6a27890 [CARRY][BUILD_GEN] Specfile updates
+9bcbe1d [CARRY][BUILD] Tooling updates
+1abdc18 [CARRY][BRANDING] Branding updates
```

Comment:
Added more context in the end of this section that will hopefully make the answer to this question clearer.